cdk/domino_cdk/config/eks.py

```diff
@@ -30,12 +30,17 @@ class NodegroupBase:
         key_name: some-key-pair - Pre-existing AWS key pair to configure for instances in the nodegorup
         min_size: 1 - Minimum node count for nodegroup. Can't be 0 on managed nodegroups.
         max_size: 10 - Maximum limit for node count in node gorup
-        machine_image: AMI and/or user_data script for nodes in nodegroup. When overriding AMI, you MUST
-                       setup your custom AMI or your user_data script _MANUALLY_ to join the cluster.
-                       Additionally, options that use user_data (ie ssm_agent) are no longer valid.
-                       machine_image:
-                         ami_id: ami-123abc
-                         user_data: ...
+        ami_id: ami-123abc - AMI to use for nodegroup, empty/null will use the default EKS AMI, which
+                             is the default. When specifying an AMI, you MUST specify a custom user_data
+                             script to join the node to the cluster, and this script must do any sort of
+                             node setup that is desired. Additionally, some nodegroup options (ie labels,
+                             taints, ssm_agent) depend on the default user data and are not supported when
+                             using custom user_data. When providing custom user_data, you must implement
```

Comment:
"depend on the default user data" -> user_data

Question: if the customer writes a script that 100% matches bootstrap.sh, can they use labels-etc?

Reply:
No, because we wouldn't know whether it was the exact same script. If we don't know what to expect, we're hands-off.

Reply:
"Depend on the default user data" *is* what I wanted to get across--they depend on the assumption that the user_data script is the default one.

Reply:
> No, because we wouldn't know

The question is not about what we know. If they provide image that has all the features of original EKS image and provide user_data that is identical to the original EKS user data (this is pretty valid scenario, most of the people do not build AMI from scratch, they take the ready one and add what they need), then labels would work. 

We just do not allow it because we cannot be sure they have all the features in AMI.

But the explanation "why" makes if more confusing. So just say "don't" and this is it.

"Additionally, some nodegroup options (ie labels, taints, ssm_agent) depend on the default user data and are not supported when using custom user_data"
-> "The nodegroup options like labels, taints and ssm_agent are not supported when using custom user_data."

Reply:
I really want to get the why in there. I gave it another shot, trying to simplify the wording. Please take another look.

Reply:
The new wording feels good to me. Closing this.